presto-parser/src/main/java/com/facebook/presto/sql/tree/GroupingOperation.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class GroupingOperation
+        extends Expression
+{
+    private final List<Expression> groupingColumns;
+
+    public GroupingOperation(Optional<NodeLocation> location, List<Expression> groupingColumns)
+    {
+        super(location);
+        requireNonNull(groupingColumns);
+        checkArgument(!groupingColumns.isEmpty(), "grouping operation columns cannot be empty");
+        checkState(groupingColumns.size() <= 63, "more than 63 grouping columns are not allowed as the resulting bit set won't fit in a Java long data type");
```

Comment:
Agreed. I've changed the line to throw an `IllegalArgumentException`, which is what other code in `presto-parser` seems to do.